### PR TITLE
Fix Auth0 expired tokens 

### PR DESCRIPTION
**Pull request by todd-elvers**
Addressed issues underlying [this Linear issue](https://linear.app/stability-ai/issue/ENG-1952/force-re-login-when-tokens-are-expired).

**Core changes:**

- The Zustand state holding the `gRPC` connection now correctly updates when the front-end receives a new access token
  - Previously the `gRPC` connection would get recomputed but Zustand would ignore the newly recomputed connection.  This issue went unnoticed for this long because our access tokens are valid for far too long, causing this to be a nasty edge case.
- Snackbars no longer overflow the screen
- Errors from remote calls can no longer overflow the screen
- `login_required` errors are no longer silently ignored
- The script to build gRPC types from proto correctly pulls down `api-interfaces`
- `Theme.Loading.Overlay` now properly consumes the entirety of the screen
- `User.Avatar` now animates when auth finishes and we have a link to the user's social avatar
- Fixed a DOM validation issue where we had a `<a>` as a child of an `<a>`

New snackbar behavior:
https://github.com/Stability-AI/platform/assets/5025641/ae8cc36f-3456-4793-8198-d411a876b814

New profile loading animation:
new-user-profile-animation

~Edit by zeke:
Completes ENG-1952

### CODE REVIEW

File: packages/app/src/Overview/index.tsx
Comment (todd-elvers) [inline packages/app/src/Overview/index.tsx:1]: This fixes a console error related to `<a> cannot be child of <a>`

File: packages/app/src/Theme/Snackbar/Provider.tsx [hunk 1]
```diff
@@ -37,33 +39,37 @@ const BaseSnackbar = React.forwardRef<
...
-          onClick={() => closeSnackbar(id)}
-        >
-          <span className="sr-only">Close</span>
-          <svg
-            aria-hidden="true"
-            className="h-5 w-5"
-            fill="currentColor"
-            viewBox="0 0 20 20"
-            xmlns="http://www.w3.org/2000/svg"
+        <div className="flex flex-grow justify-end">
```
Comment (todd-elvers) [inline packages/app/src/Theme/Snackbar/Provider.tsx:51]: Had to wrap this in a `div` to correct the `X` button's placement when `persist` was true.

File: packages/app/src/Theme/Snackbar/snackbar.scss [hunk 2]
```diff
@@ -0,0 +1,6 @@
+.notistack-SnackbarContainer {
```
Comment (todd-elvers) [inline packages/app/src/Theme/Snackbar/snackbar.scss:1]: Couldn't achieve this elegantly using Tailwind, ejected to SCSS for only these styles.

File: packages/app/src/User/Login/index.tsx [hunk 3]
```diff
@@ -72,7 +72,7 @@ export namespace Login {
         <Theme.Tooltip content="View your account page" placement="bottom">
           <Link
             to="/account/keys"
-            className="block rounded-full ring-1 ring-transparent duration-100 hover:ring hover:ring-indigo-500"
+            className="flex rounded-full ring-1 ring-transparent duration-100 hover:ring hover:ring-indigo-500"
```
Comment (todd-elvers) [inline packages/app/src/User/Login/index.tsx:75]: Fixes account icon not being centered correctly

**Review by KAJdev — APPROVED** [review]

File: packages/app/src/Theme/Snackbar/snackbar.scss (hunk 2 above)
Comment (todd-elvers) [reply packages/app/src/Theme/Snackbar/snackbar.scss:1]: oh damn, I didn't know you could do that!

**Pull request merged**